### ISSUE TITLE
Require VBO support in gl14 renderer?

### ISSUE THREAD

**Issue by AbigailBuccaneer** [colobot/colobot#1153]
**TLDR:** I think Colobot should simplify the gl14 renderer by requiring GL1.5 or vertex buffer object support. This code path is extremely old, almost no users don't need it, and its existence is holding back performance optimizations for all the other renderers. Would the colobot team consider merging a pull request that removes it? 

For context: I was working on a pull request that would reduce rendering overhead, by maintaining one vertex buffer per `EngineObject` rather than one vertex buffer per `EngineBaseObjectDataTier`. I then ran into a problem: the gl14 renderer doesn't always use vertex buffer objects, because they were added in GL 1.5. Prior to that it uses display lists, and each 'vertex buffer' is actually a display list which contains the commands to draw the entire contents of the vertex buffer.

Using display lists like this isn't compatible with the optimization I'd like to make, so I removed them. This might be a tiny performance hit for users with drivers that don't support OpenGL 1.5 or GL_ARB_vertex_buffer_object, but a win for everybody else. Then I realised, the gl14 driver now has to keep these vertex buffers around itself when vertex buffer objects aren't supported - yet more overhead for a tiny tiny hypothetical minority of people! At this point I decided that this is too much work to support a tiny fringe case. The renderer would be simpler and more flexible if it dropped support for rendering without VBO support.

Is this something the colobot team would be alright with doing? If so, I've got a nice pull request that'll improve rendering performance :innocent: 

Here's some numbers:
The [Steam hardware survey](https://store.steampowered.com/hwsurvey/) says that 98.77% of users support DirectX 10, which is roughly equivalent to GL3.3. So that's 1.23% of users across the gl14 and gl21 renderers combined.
The [0AD hardware survey](http://feedback.wildfiregames.com/report/opengl/feature/GL_ARB_vertex_buffer_object) from 2015 says that GL_ARB_vertex_buffer_object is supported by 99.8% of people (excluding those with broken drivers).

**Comment by tomangelo2** [comment]
I guess Display lists are safe to remove if that holds back improvements for the vast majority of players. For ARB removal and OpenGL upgrade - I'm also OK with that, IIRC ARB_VBO support was written because of my old GPU I don't use anymore. I don't think there are more players with such old hardware.
Related issue https://github.com/colobot/colobot/issues/222

**Comment by AbigailBuccaneer** [comment]
Closed in `dev` by c9a8a242a084667336cda9032300cdfafc45f1ab.
